IHP/ViewSupport.hs

```diff
@@ -111,6 +112,47 @@ classes !classNameBoolPairs =
 instance IsString (Text, Bool) where
     fromString string = (cs string, True)
 
+
+-- | Helper for dynamically generating attribute.
+--
+-- If given two lists, the first consisting of (key, value) tuples like
+--
+-- > [("type", "button"), ("value", "Submit")]
+--
+-- and the second consisting of (key, Bool) values like
+--
+-- > [("disabled", True)]
+--
+-- builds an attributes string consisting of all key value parts from the first list
+-- and the parts from the second list where the second value is @True@.
+--
+-- E.g.
+--
+-- >>> attributes [("type", "button"), ("value", "Submit")] ["btn", ("disabled", True)]
+-- "type=\"button\" value=\"Submit\" btn disabled"
+--
+-- >>> attributes [("type", "button"), ("value", "Submit")] ["btn", ("disabled", False)]
+-- "type=\"button\" value=\"Submit\" btn"
+--
+-- __Example:__
+--
+-- >>> <a {attributes [("role", "button")] [("hidden", True)]}>
+-- <a role="button" hidden>
+attributes :: [(Text, Text)] ->  [(Text, Bool)] -> Text
+attributes attrNameValuePairs !attrNameBoolPairs = let
+    attrNameValue =
+        attrNameValuePairs
+        |> map (\nv -> Data.Text.concat [fst nv, "=", show $ snd nv])
+
+    attrName =
+        attrNameBoolPairs
+        |> filter snd
+        |> map fst
+    in
+        unwords $ attrNameValue <> attrName
```

Comment:
I think this approach will introduce XSS problems because the attribute value might be supplied by user input. We would need to use [the blaze html `attribute` function](https://hackage.haskell.org/package/blaze-html-0.9.1.2/docs/src/Text.Blaze.Html5.Attributes.html#alt) to make it safe.

But then this function will need to return `[Attribute]` which then cannot be used like `<a {attributes [] []}>`. 

**Here are two possible solutions:**

For use cases like `disabled` or `hidden` we should IMO support boolean attribute values.  This is what you [would expect to work when coming from react's JSX](https://stackoverflow.com/a/29103418).

Then we can just write:

```html
<a role="button" hidden={True}>
```

which is a lot more clean than:

```
<a {attributes [("role", "button")] [("hidden", True)]}>
```

For this change, we would need to update the HSX compiler here: https://github.com/digitallyinduced/ihp/blob/master/IHP/HtmlSupport/QQ.hs#L118

---

Another solution for more dynamic use cases:

[In HSX we already have the following syntax](https://github.com/digitallyinduced/ihp/blob/master/IHP/HtmlSupport/Parser.hs#L76): 
```haskell
let x :: [Attributes] = ... in [hsx|<a {...x}></a>|]
```
(the syntax is implemented / reservered for future use, but the compiler part is missing, so this might fail to compile).

Maybe we can reuse this syntax to allow for:
```haskell
<a
    role="button"
    {...[ ("data-" <> getSomeString, "some-value") ]}
>
```

This is also inspired by JSX syntax.

Reply:
Good ideas, `<a href="#" hidden={False} {...[("role", role)]}>` looks much cleaner indeed!